staging/src/k8s.io/legacy-cloud-providers/vsphere/vsphere.go

```diff
@@ -1459,6 +1466,43 @@ func (vs *VSphere) NodeDeleted(obj interface{}) {
 	}
 }
 
+// Notification handler when credentials secret is added.
+func (vs *VSphere) SecretAdded(obj interface{}) {
+	klog.V(4).Infof("secret added: %+v", obj)
+	vs.refreshNodesForSecret(obj)
+}
+
+// Notification handler when credentials secret is updated.
+func (vs *VSphere) SecretUpdated(_ interface{}, newObj interface{}) {
+	klog.V(4).Infof("secret updated: %+v", newObj)
+	vs.refreshNodesForSecret(newObj)
+}
+
+// Notification handler when credentials secret is deleted.
+func (vs *VSphere) SecretDeleted(obj interface{}) {
+	klog.V(4).Infof("secret deleted: %+v", obj)
+	vs.refreshNodesForSecret(obj)
```

Comment:
What's the reason for refreshing on delete?

Reply:
The nature of the issue basically describes it. If the secret is removed, the results should be equal to situation when a `vsphere_password` env variable was removed from the config - refresh all the nodes state, and trigger events related to a misconfiguration

Reply:
Having a hard time seeing how the behavior of `GetNodeDetails` changes when this secret is deleted.

Reply:
It should fail at this point with the following error - https://github.com/Danil-Grigorev/kubernetes/blob/8a31a8921350776f0077ea17da1dbbaccd42ec35/staging/src/k8s.io/legacy-cloud-providers/vsphere/credentialmanager.go#L95

Reply:
Aaactually I think you are right. Having an additional event listener just to throw a log error message does not worth it :)